src/report.rs

```diff
@@ -6,8 +6,8 @@ use std::fmt::{Debug, Formatter};
 use parking_lot::RwLock;
 
 use crate::frames::{Frames, UnresolvedFrames};
+use crate::platform::timer::ReportTiming;
```

Comment:
Also, add a trait for `ReportTiming`, to tell the developers that it should implement `Clone`, and has a `.get_frequency()`, `.get_duration()` and `.get_start_time()` :thinking:  .

Reply:
Do the same for `validate`.